pandas/_libs/tslibs/offsets.pyx

```diff
@@ -4846,7 +4846,7 @@ cpdef to_offset(freq, bint is_period=False):
         )
 
     elif PyDelta_Check(freq):
-        return delta_to_tick(freq)
+        delta = delta_to_tick(freq)
```

Comment:
maybe on line 4841, we can do
```
    if isinstance(freq, BaseOffset):
        delta = freq
```
so then that gets validated too?

Reply:
I am unsure if it's possible. I tried to do this assignment on line 4841
```
if isinstance(freq, BaseOffset):
    delta = freq
```
but then I got failures.
The reason: if we replace `return freq` with the `delta = freq`, we go to the line 4962 and assign delta to None and then on line 4965 we raise a ValueError.
Which is why instead of the assignment `delta = freq` I added the check 
```
if is_period and not hasattr(freq, "_period_dtype_code"):
    raise ValueError(f"{freq.base} is not supported as period frequency")
```

Reply:
thanks, I saw you already made commit for it